lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module

```diff
@@ -843,14 +839,22 @@ function dosomething_campaign_is_active($node) {
    *
    * @param DateTime $date
    *  DateTime to edit
+   * @param boolean $add_four
+   *  Optional - prevents default EST datetime conversion from becoming previous day
    * @param boolean $end_of_day;
    *  Optional - set the date to end of day
    *
    * @return DateTime
    *  Updated time
    */
-function dosomething_campaign_set_est_timezone($date) {
+function dosomething_campaign_set_est_timezone($date, $add_four = FALSE, $end_of_day = FALSE) {
+  if ($add_four) {
+    $date->add(new DateInterval('PT4H'));
```

Comment:
no, if you dont add the four hours what happens is it thinks its 23:59:59 of the day beforehand because when you convert you loose four hours (its really confusing I know, damn time).

I got the code working 
```
  global $user;
    $user_timezone = new DateTimeZone($user->timezone);
    $user_date = new DateTime("now", $user_timezone);
    $new_york_timezone = new DateTimeZone('America/New_York');
    // Get the timezone difference, divide by total seconds in hour, convert to int
    $difference = (int) ($new_york_timezone->getOffset($user_date) / 3660);
```

but then I realized this is pointless! the add_four is only ever used by 'end/start date' variables because of how we modify them. this if block should actually be ignoring the user timezone.

this code might still be useful to some degree. I want to test as a user in another time zone and see if the new DateTime reflects that timezone or it defaults to UTC as well. If it defaults to UTC then the code above is sorta needed. 

If this doesnt make sense I can try explaining it again